### PR TITLE
Progress refactor

### PR DESCRIPTION
**Pull request by Gerych1984**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️

1. Adopt `Progress` to bootstrap 5.3
2. Separated widgets `Progress` and `ProgressStack`
3. Added auxiliary method "Progress::calculatedPercent()"

### CODE REVIEW

File: src/AbstractToggleWidget.php [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 abstract class AbstractToggleWidget extends Widget
 {
-    protected array $toggleOptions = [];
+    protected array $toggleOptions = array();
```
Comment (samdark) [inline src/AbstractToggleWidget.php:14]: Why? It should be `[]`.

File: src/Progress.php [hunk 2]
```diff
@@ -22,86 +21,82 @@
  * ```php
  * // default with label
  * echo Progress::widget()
- *     ->percent('60')
- *     ->label('test');
+ *     ->withPercent(60)
+ *     ->withLabel('test');
```
Comment (samdark) [inline src/Progress.php:25]: For widgets we aren't using `with*`.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/AbstractToggleWidget.php (hunk 1 above)
Comment (vjik) [reply src/AbstractToggleWidget.php:14]: Sorry, this is my experiment with Rector. I'll fix it.

**Review by vjik — COMMENTED** [review]
Fix psalm error

File: src/Progress.php [hunk 3]
```diff
@@ -136,43 +138,95 @@ public function options(array $value): self
...
     }
 
     /**
-     * Renders the progress.
      *
-     * @throws JsonException|RuntimeException if the "percent" option is not set in a stacked progress bar.
+     * @param int|float $value
+     * @param int|float $max
+     * @param int<0, max>|null $precision
+     * @param int<0, max> $mode
```
Comment (vjik) [inline src/Progress.php:160]:
```suggestion
     * @psalm-param int<0, max>|null $precision
     * @psalm-param int<0, max> $mode
```

File: src/Progress.php [hunk 4]
```diff
@@ -186,21 +240,30 @@ private function renderProgress(): string
...
-            'role' => 'progressbar',
-            'aria-valuenow' => $percent,
-            'aria-valuemin' => 0,
-            'aria-valuemax' => 100,
-        ]);
+        if ($this->animated) {
+            $classNames[] = 'progress-bar-animated';
+        }
 
         /** @psalm-suppress InvalidArgument */
```
Comment (vjik) [inline src/Progress.php:258]:
```suggestion
```

File: src/ProgressStack.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace Yiisoft\Yii\Bootstrap5;
+
+use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Html\Html;
+
+use function array_map;
+
+/**
+ * ProgressStack renders a bootstrap progress bar component.
```
Comment (vjik) [inline src/ProgressStack.php:13]:
```suggestion
 * `ProgressStack` renders a bootstrap progress bar component.
```

File: src/ProgressStack.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        $options = $this->options;
+
+        if (!isset($options['id'])) {
+            $options['id'] = $this->getId();
+        }
+
+        $tag = ArrayHelper::remove($options, 'tag', 'div');
+
+        /** @psalm-suppress InvalidArgument */
+        Html::addCssClass($options, ['progress-stacked']);
```
Comment (vjik) [inline src/ProgressStack.php:56]:
```suggestion
        Html::addCssClass($options, 'progress-stacked');
```

File: tests/ProgressTest.php [hunk 7]
```diff
@@ -17,53 +19,229 @@ public function testRender(): void
...
+                ]),
+            Progress::widget()
+                ->id('bar-3')
+                ->percent(20)
+                ->barOptions([
+                    'class' => 'bg-info',
+                ]),
+        ];
+
+        $html = ProgressStack::widget()
```
Comment (vjik) [inline tests/ProgressTest.php:201]: Move `ProgressStack` tests to `ProgressStackTest` class.

**Review by vjik — CHANGES_REQUESTED** [review]

File: .php-cs-fixer.cache
Comment (vjik) [inline .php-cs-fixer.cache:1]: remove it

**Review by vjik — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by vjik** [comment]
@Gerych1984 Thanks 👍

**Pull request merged**